test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/CSharpRazorCodeGeneratorTest.cs

```diff
@@ -404,6 +404,11 @@ public void CSharpChunkGeneratorCorrectlyGeneratesDesignTimePragmasMarkupAndExpr
             });
         }
 
+        [Fact]
+        public void CSharpChunkGeneratorCorrectlyGeneratesDesignTimePragmasForExplicitExpressionContainingMarkup()
```

Comment:
Should do a runtime test as well to ensure all is said and done. We don't currently test this scenario.